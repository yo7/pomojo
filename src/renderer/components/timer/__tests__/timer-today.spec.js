import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import TimerToday from '../timer-today.vue'

describe('TimerToday', () => {
  let wrapper

  it('contains text when it has today count and goal count', () => {
    wrapper = shallow(TimerToday, {propsData: {
      resting: false,
      today: 4,
      goal: 8
    }})
    expect(wrapper.contains('.text')).toBe(true)
  })

  it('does not contain text when the goal count is 0', () => {
    wrapper = shallow(TimerToday, {propsData: {
      resting: false,
      today: 4,
      goal: 0
    }})
    expect(wrapper.contains('.text')).toBe(false)
  })

  it('contains today and goal text', () => {
    wrapper = shallow(TimerToday, {propsData: {
      resting: false,
      today: 2,
      goal: 8
    }})
    expect(wrapper.find('.text').text().trim()).toBe('2/8')
  })

  it('renders correctly', () => {
    const renderer = createRenderer()
    wrapper = shallow(TimerToday, {propsData: {
      resting: false,
      toay: 4,
      goal: 8
    }})
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error()
      }
      expect(str).toMatchSnapshot()
    })
  })
})
