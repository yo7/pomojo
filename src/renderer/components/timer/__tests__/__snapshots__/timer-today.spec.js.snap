// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerToday renders correctly 1`] = `
<div class="timer-today">
    <div class="text">
        4/8
    </div> <progress max="100" value="50" class="progress"></progress></div>
`;
